executor/common_linux.h

```diff
@@ -5655,3 +5655,23 @@ static void setup_swap()
 }
 
 #endif
+
+#if SYZ_EXECUTOR || __NR_syz_pidfd_open
+#include <sys/syscall.h>
+
+static long syz_pidfd_open(volatile long pid, volatile long flags)
+{
+	if (pid == 1) {
+		// Under a PID namespace, pid=1 is the parent process.
+		// We don't want a forked child to mangle parent syz-executor's fds.
+		pid = 0;
+	}
+#ifdef SYS_pidfd_open
```

Comment:
We need to use the real syscall number always.
The libc headers where executor built are unrelated to the tested kernel.
And if the syscall is not supported, syz_pidfd_open should return ENOSYS rather than 0.
Either define the const if it's not defined, or maybe we pass these from our descriptions somewhere already.

Reply:
We use `syscall(SYS_...)` in many other places in `common_linux.h`. Should that also be somehow addressed?

What makes it harder is that syscall numbers actually depend on the arch, so by doing
```
#define  SYS_pidfd_open 434
```

we risk calling some different syscall on non-x86. For flag definitions we probably don't care that much, but with syscalls it can get tricky. 

So we either build lists of `#ifdef ... #elif ... #endif` it for every such const in `common_linux.h`, or we need to somehow substitute these `SYS_` consts into the `syz-executor` code like we do it for the `#if !SYZ_EXECUTOR` case in `pkg/csource` (?).

Reply:
> We use syscall(SYS_...) in many other places in common_linux.h. Should that also be somehow addressed?

This is fine. If it's defined, it's the right one.

I see we already define __NR_ consts in executor/defs.h, use these.

Reply:
> I see we already define _NR consts in executor/defs.h, use these.

But these don't get included into C repros, do they?

Reply:
>  If it's defined, it's the right one.

In this case, it's defined in newer Linux headers and not defined in the older ones (that we check in `ci / old`).

Reply:
> But these don't get included into C repros, do they?

These should be included, otherwise compilation of repros will fail if they use newest syscalls.

> In this case, it's defined in newer Linux headers and not defined in the older ones (that we check in ci / old).

Doing `syscall(SYS_...)` is correct.
It will either compile and work as expected, or fail to compile (but these failures are loud and thus are known to not happen in real life).

Doing
```
#ifdef SYS_...
  syscall(SYS_...);
#else
  ...
#endif
```
is wrong because it will always compile, but it may do wrong thing depending on what machine it's compiled.
Behavior of the executor should not depend on the host machine.

Reply:
> These should be included, otherwise compilation of repros will fail if they use newest syscalls.

FTR we add a `#define {}_NR {}` for every call in the program in `pkg/csource` here. We don't reply on `defs.h`.

https://github.com/google/syzkaller/blob/b455167d16775e043bcb382a6281082ff0b86a6f/pkg/csource/csource.go#L214-L229

Reply:
I think we can reuse `PseudoSyscallDeps` for this purpose.

Reply:
Done. PTAL.